File: Java/1329. Sort the Matrix Diagonally.java
class Solution {
    public int[][] diagonalSort(int[][] mat) {
        for(int i=0;i<mat[0].length;i++)
            sort(mat,0,i);
        for(int i=0;i<mat.length;i++)
            sort(mat,i,0);
        return mat;
    }
    public void sort(int[][] mat,int row,int column)
    {
        int r=row;
        int c=column;
        List<Integer> list=new ArrayList<>();
        while(r<mat.length&& c<mat[0].length)
        {
            list.add(mat[r][c]);
            r++;
            c++;
        }
        Collections.sort(list);
        r=row;
        c=column;
        int index=0;
        while(r<mat.length && c<mat[0].length)
        {
            mat[r][c]=list.get(index);
            index++;
            r++;
            c++;
        }
    }
}

File: Python/1329. Sort the Matrix Diagonally.py
class Solution:
    def diagonalSort(self, mat: List[List[int]]) -> List[List[int]]:
        a1 = []
        a2 = []
        i = 0
        
        while i < len(mat[0]):
            a = 0
            b = i
            tmp = []
            while a < len(mat) and b < len(mat[0]):
                tmp.append(mat[a][b])
                a+=1
                b+=1
                
            a1.append(tmp)
            i+=1
            
        i = 1
        while i < len(mat):
            a = i
            b = 0
            tmp = []
            while a < len(mat) and b < len(mat[0]):
                tmp.append(mat[a][b])
                a+=1
                b+=1
            a2.append(tmp)
            i+=1
            
        for i in a1:
            i.sort()
        for i in a2:
            i.sort()
        i = 0
        d = 0
        while i < len(mat[0]):
            a = 0
            b = i
            k = 0
            while a < len(mat) and b < len(mat[0]):
                mat[a][b] = a1[d][k]
                a+=1
                b+=1
                k+=1
            
            i+=1
            d+=1  
        i = 1
        d = 0
        while i < len(mat):
            a = i
            b = 0
            k = 0
            while a < len(mat) and b < len(mat[0]):
                mat[a][b] = a2[d][k]
                a+=1
                b+=1
                k+=1
            
            i+=1
            d+=1
        
        return mat